In [177]:
%matplotlib inline
import numpy as np
import pandas as pd
import scipy 
import matplotlib.pyplot as plt 

In [178]:
md = pd.read_csv('archive/movies_metadata.csv')
md.head().transpose()

/Users/lucasdu/Documents/aut20_DataScience/uc_datascience/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3147: DtypeWarning: Columns (10) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


0  \
adult                                                              False   
belongs_to_collection  {'id': 10194, 'name': 'Toy Story Collection', ...   
budget                                                          30000000   
genres                 [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
homepage                            http://toystory.disney.com/toy-story   
id                                                                   862   
imdb_id                                                        tt0114709   
original_language                                                     en   
original_title                                                 Toy Story   
overview               Led by Woody, Andy's toys live happily in his ...   
popularity                                                       21.9469   
poster_path                             /rhIRbceoE9lR4veEXuwCC2wARtG.jpg   
production_companies      [{'name': 'Pixar Animation Studios', 'id': 3}]   
production_countries   [{'iso_3166_1': 'US', 'name': 'United States o...   
release_date                                                  1995-10-30   
revenue                                                      3.73554e+08   
runtime                                                               81   
spoken_languages                [{'iso_639_1': 'en', 'name': 'English'}]   
status                                                          Released   
tagline                                                              NaN   
title                                                          Toy Story   
video                                                              False   
vote_average                                                         7.7   
vote_count                                                          5415   

                                                                       1  \
adult                                                              False   
belongs_to_collection                                                NaN   
budget                                                          65000000   
genres                 [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
homepage                                                             NaN   
id                                                                  8844   
imdb_id                                                        tt0113497   
original_language                                                     en   
original_title                                                   Jumanji   
overview               When siblings Judy and Peter discover an encha...   
popularity                                                       17.0155   
poster_path                             /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg   
production_companies   [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
production_countries   [{'iso_3166_1': 'US', 'name': 'United States o...   
release_date                                                  1995-12-15   
revenue                                                      2.62797e+08   
runtime                                                              104   
spoken_languages       [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
status                                                          Released   
tagline                        Roll the dice and unleash the excitement!   
title                                                            Jumanji   
video                                                              False   
vote_average                                                         6.9   
vote_count                                                          2413   

                                                                       2  \
adult                                                              False   
belongs_to_collection  {'id': 119050, 'name': 'Grumpy Old Men Collect...   
budget                                   

In [179]:
md.columns

Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')

In [180]:
md.shape

(45466, 24)

In [181]:
md.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

In [182]:
md = md.drop(['imdb_id'], axis=1)
md = md.drop(['homepage'], axis=1)
md = md.drop(['tagline'], axis=1)

In [183]:
md[md['adult'] == 'False'].shape

(45454, 21)

We see that 45454 movies have False in the adult column (so only 12 movies are adult movies), which means we can safely discard this column as well.

In [184]:
md = md.drop(['adult'], axis=1)

In [185]:
md[md['status'] == 'Released'].shape

(45014, 20)

In [186]:
md = md.drop(['status'], axis=1)

In [187]:
md = md.drop(['original_title'], axis=1)
md = md.drop(['video'], axis=1)

In [188]:
md[md['revenue'] == 0].shape

(38052, 17)

In [189]:
md['revenue'] = md['revenue'].replace(0, np.nan)
md['original_language'] = md['original_language'].replace('xx', np.nan)

Now, we also convert the belongs_to_collection column to a binary of 1 or 0; 1 if the movie does belong to a collection, 0 if it does not.

In [190]:
md['belongs_to_collection'] = md['belongs_to_collection'].fillna('').apply(lambda x: 0 if x=='' else 1)

In [191]:
md['budget'] = pd.to_numeric(md['budget'], errors='coerce')
md['budget'] = md['budget'].replace(0, np.nan)

In [192]:
md = md.drop(['poster_path'], axis=1)

Now, we remove all rows with a NaN value, leaving only the movies that have complete data.

In [193]:
md = md.dropna()

In [194]:
md.to_csv('clean_metadata.csv')

In [195]:
md.head().transpose()

0  \
belongs_to_collection                                                  1   
budget                                                             3e+07   
genres                 [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
id                                                                   862   
original_language                                                     en   
overview               Led by Woody, Andy's toys live happily in his ...   
popularity                                                       21.9469   
production_companies      [{'name': 'Pixar Animation Studios', 'id': 3}]   
production_countries   [{'iso_3166_1': 'US', 'name': 'United States o...   
release_date                                                  1995-10-30   
revenue                                                      3.73554e+08   
runtime                                                               81   
spoken_languages                [{'iso_639_1': 'en', 'name': 'English'}]   
title                                                          Toy Story   
vote_average                                                         7.7   
vote_count                                                          5415   

                                                                       1  \
belongs_to_collection                                                  0   
budget                                                           6.5e+07   
genres                 [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
id                                                                  8844   
original_language                                                     en   
overview               When siblings Judy and Peter discover an encha...   
popularity                                                       17.0155   
production_companies   [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
production_countries   [{'iso_3166_1': 'US', 'name': 'United States o...   
release_date                                                  1995-12-15   
revenue                                                      2.62797e+08   
runtime                                                              104   
spoken_languages       [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
title                                                            Jumanji   
vote_average                                                         6.9   
vote_count                                                          2413   

                                                                       3  \
belongs_to_collection                                                  0   
budget                                                           1.6e+07   
genres                 [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
id                                                                 31357   
original_language                                                     en   
overview               Cheated on, mistreated and stepped on, the wom...   
popularity                                                       3.85949   
production_companies   [{'name': 'Twentieth Century Fox Film Corporat...   
production_countries   [{'iso_3166_1': 'US', 'name': 'United States o...   
release_date                                                  1995-12-22   
revenue                                                      8.14522e+07   
runtime                                                              127   
spoken_languages                [{'iso_639_1': 'en', 'name': 'English'}]   
title                                                  Waiting to Exhale   
vote_average                                                         6.1   
vote_count                                                            34   

                                                                       5  \
belongs_to_collection                                                  0   
budget                                  

In [196]:
md.shape

(5368, 16)

In [197]:
md.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5368 entries, 0 to 45422
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   belongs_to_collection  5368 non-null   int64  
 1   budget                 5368 non-null   float64
 2   genres                 5368 non-null   object 
 3   id                     5368 non-null   object 
 4   original_language      5368 non-null   object 
 5   overview               5368 non-null   object 
 6   popularity             5368 non-null   object 
 7   production_companies   5368 non-null   object 
 8   production_countries   5368 non-null   object 
 9   release_date           5368 non-null   object 
 10  revenue                5368 non-null   float64
 11  runtime                5368 non-null   float64
 12  spoken_languages       5368 non-null   object 
 13  title                  5368 non-null   object 
 14  vote_average           5368 non-null   float64
 15  vot

In [198]:
md[md['original_language']=='xx'].shape

(0, 16)